subsys/caf/modules/buttons.c

```diff
@@ -486,8 +480,17 @@ static void button_pressed_isr(const struct device *gpio_dev,
 
 static void button_pressed_fn(struct k_work *work)
 {
-	int err = callback_ctrl(0);
+	int err;
+
+	/* Scanning may be scheduled, switch off pins */
```

Comment:
I would suggest to stick with `/* Scanning will be scheduled, switch off pins */` (suspended system is expected to wakeup and schedule scanning)

Reply:
In the commit message you could improve: `GPIO API does not allow for configuring or setting GPIOs from ISR context.`